porcupine/plugins/filetypes.py

```diff
@@ -252,6 +253,20 @@ def parse_filetype_name(name: str) -> FileType:
     )
 
 
+filetypes_var: tkinter.StringVar | None = None
+
+
+def _sync_filetypes_menu(event: object) -> None:
+    tab = get_tab_manager().select()
+    if isinstance(tab, tabs.FileTab) and tab.path is not None:
```

Comment:
We shouldn't check `tab.path`, because not-yet-saved files can also have a filetype. We should also ensure that no menu item shows up as selected when the tab isn't a `FileTab`. For example, Porcupine debug prompt tabs are not `FileTab`s, and no filetype should show up as selected when you are in the debug prompt.